src/xdp-session.c

```diff
@@ -311,6 +311,20 @@ xdp_session_authorize_callback (GDBusInterfaceSkeleton *interface,
   return TRUE;
 }
 
+static gboolean
+is_valid_token(const char *token)
+{
+  int i;
+
+  for (i = 0; token[i]; i++)
+    {
+      if (!g_ascii_isalnum(token[i]) && token[i] != '_')
+        return FALSE;
+    }
+
+  return TRUE;
```

Comment:
This implementation would return true for the empty string, but `"/blah/blah/" + ""` is not a syntactically valid object path (an object path must not end with `/` unless it is exactly `/`). So we must check for that.

(We are probably appending it to some string in the `/org/freedesktop/...` namespace rather than literally `/foo/` or `/blah/blah/`, but it's the "shape" that matters.)